android/shopify_buy_plugin/src/main/java/com/shopify/unity/buy/AndroidPayCheckoutSession.java

```diff
@@ -15,22 +17,45 @@ public AndroidPayCheckoutSession() {
         this.logger = new AndroidLogger();
     }
 
-    public AndroidPayCheckoutSession(ILogger logger) {
+    AndroidPayCheckoutSession(ILogger logger) {
         this.logger = logger;
     }
 
+    //CHECKSTYLE:OFF
     public boolean checkoutWithAndroidPay(
             String unityDelegateObjectName,
             String merchantName,
+            String publicKey,
             String pricingLineItemsString,
             String currencyCode,
             String countryCode,
-            boolean requiresShipping
+            boolean requiresShipping,
+            boolean testing
     ) {
+        //CHECKSTYLE:ON
+        if (!PayHelper.isAndroidPayEnabledInManifest(UnityPlayer.currentActivity)) {
+            // TODO: Send unsupported error to Unity
+            return false;
+        }
+
         try {
             PayCart cart = cartFromUnity(merchantName, pricingLineItemsString, currencyCode, countryCode,
                     requiresShipping);
-            addAndroidPayFragment(unityDelegateObjectName, cart, countryCode);
+
+            UnityAndroidPayFragment payFragment = UnityAndroidPayFragment.builder()
```

Comment:
Since we add the fragment in one place I moved this inline with this method since this method isn't long anyways.